feat: add CP FE

Mikhail-Sosnin opened the pull request.

DenisVerkhoturov committed:
chore: Initialize the project

DenisVerkhoturov committed:
feat: Provide initial component with styles

DenisVerkhoturov committed:
chore: Add `prettier` as dev dependency and configure it

DenisVerkhoturov committed:
feat: Add login page mock-up

DenisVerkhoturov committed:
chore: Add git hook lifecycle scripts

DenisVerkhoturov reviewed (changes requested):
Thank you, @Mikhail-Sosnin, however, I would rather upload code from my machine since there I did more. Just give me a little time.

Mikhail-Sosnin commented:
> Thank you, @Mikhail-Sosnin, however, I would rather upload code from my machine since there I did more. Just give me a little time.

yep, sure
feel free to use forcepush in this branch

DenisVerkhoturov committed:
chore: Bump `parcel-bundler` version

Security mainanance, fixes transitive moderate vulnerability.

Resolves: https://www.npmjs.com/advisories/790

DenisVerkhoturov committed:
chore: Give proper name for the project

DenisVerkhoturov committed:
chore: Add multistage Dockerfile

DenisVerkhoturov committed:
feat: Add `/sign-in` and `/sign-up` pages components

DenisVerkhoturov commented:
My bad, I accidentally closed the pull-request when pushed unrelated history. However, this pull request now contains all the changes. Will work with it a bit later if everybody doesn't mind.

Thinks to fix:
* The application in the built `Dockerfile` unable to access some files - looks like the problem is about the way the artefacts are built (to check `parcel`)
* Even `production` artefacts contain source maps - the problem is `typescript` configuration `json` (to make source maps generate only in the development mode and not in the production mode)
* Some dependencies look redundant - looks like at some point I experimented with different libraries and have forgotten to clean up)
